### PR TITLE
CIN-06051: Ensure t.null is never added to a query

### PR DESCRIPTION
**Pull request by cinchy-yogeshjagtap**
-- Added Info message "The columns "Child Form Link Field" or both the “Child Form Parent ID” and the "Child Form Link ID" must be populated in your form fields table. " when Child Form Link Field or Child Form Parent ID and the Child Form Link ID column data is empty.

### CODE REVIEW

**Review by cinchy-m — DISMISSED** [review]

File: src/app/dynamic-forms/service/form-helper/form-helper.service.ts [hunk 1]
```diff
@@ -280,6 +280,10 @@ export class FormHelperService {
     }
 
     const selectQuery: IQuery = form.generateSelectQuery(targetRowId);
+    
+    if(isNullOrUndefined(selectQuery)){
```
Comment (DAquilinaCinchy) [inline src/app/dynamic-forms/service/form-helper/form-helper.service.ts:284]: add space after `if` and before `{`

File: src/app/dynamic-forms/models/cinchy-form.model.ts [hunk 2]
```diff
@@ -826,7 +826,7 @@ export class Form {
         null
       );
 
-      return query;
+      return this.childFormLinkId != null ? query : null;
```
Comment (DAquilinaCinchy) [inline src/app/dynamic-forms/models/cinchy-form.model.ts:829]: There's no reason to do the work of building the query if we're just going to discard it based on a boolean condition. Just add `if (!this.childFormLinkId)` on line 815 and return null if it's truthy

File: src/app/dynamic-forms/fields/child-form-table/child-form-table.component.html [hunk 3]
```diff
@@ -49,7 +49,13 @@
       </tbody>
     </table>
   </div>
-  <p *ngIf="!childForm.childFormRowValues?.length">
-    There is no data in this table. You can use the add button above to get started.
+  <p *ngIf="childForm.childFormLinkId == null || childForm.childFormParentId == null ;else noRecord">
```
Comment (DAquilinaCinchy) [inline src/app/dynamic-forms/fields/child-form-table/child-form-table.component.html:52]: This isn't a valid condition. The purpose of this message was to show an empty state for when there are no records to display, but in that case you're still able to create new entries. If the child form isn't configured correctly (which is the condition you're checking) then you need to either show something else or just not render the entire form

**Review by DAquilinaCinchy — CHANGES_REQUESTED** [review]

File: src/app/dynamic-forms/fields/child-form-table/child-form-table.component.html (hunk 3 above)
Comment (cinchy-m) [reply src/app/dynamic-forms/fields/child-form-table/child-form-table.component.html:52]: @DAquilinaCinchy FYI, I can't comment on the code but this appears to work exactly as expected. If the criteria is met (required columns have values) then the "There is no data in this table. You can use the add button above to get started" is displayed. If not, "The columns "Child Form Link Field" or both the “Child Form Parent ID” and the "Child Form Link ID" must be populated in your form fields table." is displayed.

File: src/app/dynamic-forms/models/cinchy-form.model.ts (hunk 2 above)
Comment (cinchy-yogeshjagtap) [reply src/app/dynamic-forms/models/cinchy-form.model.ts]: Hi @DAquilinaCinchy , I have resolved above comment.

File: src/app/dynamic-forms/fields/child-form-table/child-form-table.component.html [hunk 4]
```diff
@@ -49,7 +49,13 @@
       </tbody>
     </table>
   </div>
-  <p *ngIf="!childForm.childFormRowValues?.length">
-    There is no data in this table. You can use the add button above to get started.
+  <p *ngIf="childForm.childFormLinkId == null || childForm.childFormParentId == null ;else noRecord"> 
+     The columns "Child Form Link Field" or both the “Child Form Parent ID” and the "Child Form Link ID" must be populated in your form fields table. 
   </p>
```
Comment (cysonya) [inline src/app/dynamic-forms/fields/child-form-table/child-form-table.component.html:54]: The quotes around "Child Form Parent ID" are different, this usually happens when copy/pasting.  We should maybe change the quotes to be the same as others.

Comment (cinchy-yogeshjagtap) [reply src/app/dynamic-forms/fields/child-form-table/child-form-table.component.html]: Hi @cysonya , I have resolved above comment.

**Review by DAquilinaCinchy — APPROVED** [review]

**Review by cysonya — APPROVED** [review]

**Pull request merged**